source/common/network/connection_impl.cc

```diff
@@ -284,9 +284,19 @@ void ConnectionImpl::setBufferLimits(uint32_t limit) {
   // asynchronously, we have the option of either setting the watermarks aggressively, and regularly
   // enabling/disabling reads from the socket, or allowing more data, but then not triggering
   // based on watermarks until 2x the data is buffered in the common case.  Given these are all soft
-  // limits we err on the side of buffering more and having better performace.
+  // limits we err on the side of buffering more triggering watermark callbacks less often.
```

Comment:
"buffering more and ..."

OK this makes sense to me now, thanks. I guess my follow up question is whether this is needed though. The assumption is that we will be writing limit bytes in the typical case. I kind of doubt that will happen (limits would typically be set with some margin above the typical back and forth sizes). Given that, does the extra complexity and groking time provide enough value? (I don't really feel that strongly about it, mostly asking the question).